core/src/main/java/org/apache/iceberg/BaseUpdatePartitionSpec.java

```diff
@@ -192,6 +211,12 @@ public BaseUpdatePartitionSpec removeField(Term term) {
 
   @Override
   public BaseUpdatePartitionSpec renameField(String name, String newName) {
+    PartitionField existingField = nameToField.get(newName);
+    if (existingField != null && isVoidTransform(existingField)) {
+      // rename the old deleted field that is being replaced by the new field
+      renameField(existingField.name(), existingField.name() + "_" + UUID.randomUUID());
```

Comment:
Why use `UUID.randomUUID()` instead of the partition field ID? I think it makes more sense to use `existingField.fieldId()`.

Reply:
change to `existingField.fieldId()`